server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -276,14 +297,43 @@ public void parse(ParseContext context) throws IOException {
                 context.doc().add(field);
             }
         } catch (Exception e) {
-            throw new MapperParsingException("failed to parse field [{}] of type [{}] in document with id '{}'", e, fieldType().name(),
-                    fieldType().typeName(), context.sourceToParse().id());
+            String valuePreview = "";
+            try {
+                XContentParser parser = context.parser();
+                if (parser.currentToken().isValue()) {
```

Comment:
I think we can remove these conditions, and only use `AbstractXContentParser.readValue` here